### PR TITLE
feat(core,kafka): Detailed ingestion logging for matching subset of records

### PR DESCRIPTION
**Pull request by velvia**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**New behavior :**

- Filter for `TestConsumer` to dump out raw Kafka ingestion records when a string partition key field matches a given value
- A new dataset/store config that when enabled allows detailed logging of each sample ingested when the partition key has fields matching the config

### CODE REVIEW

File: core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala [hunk 1]
```diff
@@ -392,6 +392,34 @@ extends ChunkMap(memFactory, initMapSize) with ReadablePartition {
...
+ */
+class TracingTimeSeriesPartition(partID: Int,
+                                 dataset: Dataset,
+                                 partitionKey: BinaryRegion.NativePointer,
+                                 shard: Int,
+                                 bufferPool: WriteBufferPool,
+                                 shardStats: TimeSeriesShardStats,
+                                 memFactory: MemFactory,
+                                 initMapSize: Int) extends
+TimeSeriesPartition(partID, dataset, partitionKey, shard, bufferPool, shardStats, memFactory, initMapSize) {
```
Comment (vishramachandran) [inline core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala]: Prefer composition /  decorator pattern over inheritance.  
Inheritance leads to bad problems later on (remember Stack extends from Vector in java) ? One such place we already have  in code is OnDemandPagingShard which should have not extended TimeSeriesShard.

However, I dont have an alternate solution with less code changes without adding more memory usage for TSP. :(

Please add a comments so readers understand.

File: conf/timeseries-dev-source.conf [hunk 2]
```diff
@@ -72,6 +72,11 @@
         # Thats about 6MB
         evicted-pk-bloom-filter-capacity = 50000
 
+        # Uncomment to log at DEBUG ingested samples matching certain filters
+        # Only works for StringColumn fields in the partition key
+        # trace-filters = {
+        #   metric = "bad-metric-to-log"
```
Comment (vishramachandran) [inline conf/timeseries-dev-source.conf]: Should we include _ns in the filter as well?
If the filter needs to be on a popular metric name, too much debug  can happen.

File: core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala (hunk 1 above)
Comment (velvia) [reply core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala]: The reason why I used inheritance was not so much memory but just ease of implementation.  With composition we'd need to add in tons of methods and clutter things up quite a bit.  If it simply implemented `ReadablePartition` that might break things in a bunch of places.  So best way to keep changes small and balance out different needs.

File: conf/timeseries-dev-source.conf (hunk 2 above)
Comment (velvia) [reply conf/timeseries-dev-source.conf]: For the current debug case, `StringColumn` filtering is fine, we don't need filtering by Map/labels.  We can extend it later if that is needed.  Let's keep it simple.

**Pull request merged**